CMakeLists.txt

```diff
@@ -73,6 +73,7 @@ endif()
 
 
 include_directories(${CMAKE_BINARY_DIR})  # Make the exv_conf.h file visible for the full project
+include_directories( BEFORE SYSTEM include) # for  #include <exiv2/exiv2.hpp>
```

Comment:
In my opinion it is wrong to do this. We already have CMake code dealing with the inclusion of the include directory in the file `src/CMakeLists.txt`:

```cmake
# Use include/exiv2 to build library code
target_include_directories(exiv2lib PUBLIC
    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include/exiv2>  # 1
    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>           # 2
    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>         # 3
)
```

I'll explain the purpose of including each of those lines:
```cpp
// #1 is used so that the .cpp files which have includes like the following can find the headers while building the library:

// basicio.cpp
// included header files
#include "config.h"
#include "datasets.hpp"
#include "basicio.hpp"
#include "futils.hpp"
#include "types.hpp"
#include "error.hpp"
#include "http.hpp"
#include "properties.hpp"

// #2 is used so that the .cpp files which have includes like the following can find the headers while building the library or apps:

// exiv2.cpp
#include <exiv2/exiv2.hpp>

// #3 is used so that other projects consuming exiv2 will automatically include the right include directories when consuming the library via CMake.
```
I would even take an additional step and remove `#1` so that all the inclusions to headers are done in the same way.